openedx_tagging/core/tagging/api.py

```diff
@@ -315,3 +309,40 @@ def autocomplete_tags(
         # remove repeats
         .distinct()
     )
+
+
+def add_tag_to_taxonomy(
+    taxonomy: Taxonomy,
+    tag: str,
+    parent_tag_id: int | None = None,
+    external_id: str | None = None
+) -> Tag:
+    """
+    Adds a new Tag to provided Taxonomy. If a Tag already exists in the
+    Taxonomy, an exception is raised, otherwise the newly created
+    Tag is returned
+    """
+    return taxonomy.cast().add_tag(tag, parent_tag_id, external_id)
+
+
+def update_tag_in_taxonomy(taxonomy: Taxonomy, tag: int, tag_value: str):
```

Comment:
Ooh good points, @bradenmacdonald -- I agree that using `value` where we can makes sense.

Reply:
I see, that makes sense, I updated the implementation to use values instead of IDs. https://github.com/openedx/openedx-learning/commit/3abce26a6fdd11f89dd4493db79a4d41a008c7c3